# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsIFrame.cpp
# Commit: 20a5c1f3b35e
# Full Hash: 20a5c1f3b35e80849e0b98ffbe3f9488d703717b
# Author: Frederic Wang <fred.wang@free.fr>
# Date: 2023-11-22 21:50:37
# Regressor Bug: 1807253
# File Overlap Count: 1
# Description:
#   Bug 1807253 - unreliable timing for content-visibility:auto, r=emilio
#   
#   Rewrite implementation of content-visibility: auto as defined in
#   https://github.com/w3c/csswg-drafts/issues/8542
#   
# ==============================================================================

diff -r b90083f581e1 -r 20a5c1f3b35e layout/generic/nsIFrame.cpp
--- a/layout/generic/nsIFrame.cpp	Mon Nov 20 11:33:16 2023 +0000
+++ b/layout/generic/nsIFrame.cpp	Wed Nov 22 13:00:36 2023 +0000
@@ -766,9 +766,6 @@
   if (StyleDisplay()->ContentVisibility(*this) ==
       StyleContentVisibility::Auto) {
     PresShell()->RegisterContentVisibilityAutoFrame(this);
-    auto* element = Element::FromNodeOrNull(GetContent());
-    MOZ_ASSERT(element);
-    PresContext()->Document()->ObserveForContentVisibility(*element);
   } else if (auto* element = Element::FromNodeOrNull(GetContent())) {
     element->ClearContentRelevancy();
   }
@@ -840,13 +837,6 @@
     }
   }
 
-  if (StyleDisplay()->ContentVisibility(*this) ==
-      StyleContentVisibility::Auto) {
-    if (auto* element = Element::FromNodeOrNull(GetContent())) {
-      PresContext()->Document()->UnobserveForContentVisibility(*element);
-    }
-  }
-
   // Disable visibility tracking. Note that we have to do this before we clear
   // frame properties and lose track of whether we were previously visible.
   // XXX(seth): It'd be ideal to assert that we're already marked nonvisible